src/utils/common.ts

```diff
@@ -262,11 +264,13 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
-  let nativeGasCost = await voidSigner.estimateGas({
-    ...unsignedTx,
-    gasPrice,
-    gasLimit,
-  });
+  let nativeGasCost = gasUnits
+    ? BigNumber.from(gasUnits)
+    : await voidSigner.estimateGas({
+        ...unsignedTx,
+        gasPrice,
+        gasLimit,
+      });
```

Comment:
ooc, do you think it's a hard requirement to be able to specify `gasLimit` and `gasUnits` independently?

(I think the historical reason for needing to override gasLimit was probably debunked and we fortunately haven't hit another issue...at least as far as I am aware)

Reply:
Yea, that's a good point. We don't use that anywhere. I can remove that parameter then

Reply:
https://github.com/across-protocol/sdk/pull/717/commits/e662f595ec43410d79cf4c81be8414e6fa474cd5